src/util/simplify_expr.cpp

```diff
@@ -135,25 +135,22 @@ bool simplify_exprt::simplify_popcount(popcount_exprt &expr)
 
   if(op.is_constant())
   {
-    const typet &type=ns.follow(op.type());
+    const typet &op_type = op.type();
 
-    if(type.id()==ID_signedbv ||
-       type.id()==ID_unsignedbv)
+    if(op_type.id() == ID_signedbv || op_type.id() == ID_unsignedbv)
     {
-      mp_integer value;
-      if(!to_integer(op, value))
-      {
-        std::size_t result;
+      const auto width = to_bitvector_type(op_type).get_width();
```

Comment:
How about:
```
const auto value = numeric_cast<mp_integer>(op);
if(!value.has_value())
  return true;

std::size_t result = 0;
while(*value != 0)
{
  ++result;
  *value = bitwise_and(*value, *value - 1);
}

exprt simp_result = from_integer(result, expr.type());
expr.swap(simp_result);

return false;
```

But I don't know which of them is faster in practice, given `bitwise_and` may be expensive.

Reply:
Ok, thanks!